src/AttributesAndMethods.jl

```diff
@@ -519,115 +79,26 @@ export mori_cone
-
-Compute the coordinate ring of the torus factor of the normal toric variety `v`, using `names` as label for the homogeneous coordinates.
-"""
-function coordinate_ring_of_torus( v::AbstractNormalToricVariety, names::Vector{String} )
-    gap_names = [ GapObj( names[ i ] ) for i in 1 : size(names)[1] ]
-    gap_names = GapObj( gap_names )
-    gap_CoordinateRingOfTorus = GAP.Globals.CoordinateRingOfTorus( v.GapNTV, gap_names )
-    return CoordinateRingOfTorus( gap_CoordinateRingOfTorus )
-end
-export coordinate_ring_of_torus
-
-
-@doc Markdown.doc"""
-    cox_ring( v::AbstractNormalToricVariety, name::String )
-
-Compute the Cox ring of the normal toric variety `v`, using `name` as label for the homogeneous coordinates.
-"""
-function cox_ring( v::AbstractNormalToricVariety, names::String )
-    gap_names = GapObj( names )
-    gap_CoxRing = GAP.Globals.CoxRing( v.GapNTV, gap_names )
-    return CoxRing( gap_CoxRing )
-end
-export cox_ring
-
-
-@doc Markdown.doc"""
-    Base.:*( v::AbstractNormalToricVariety, w::AbstractNormalToricVariety )
-
-Compute the direct product of the normal toric varieties `v` and `w`.
-
-# Examples
-```jldoctest
-julia> projective_space( 2 ) * projective_space( 2 )
-A normal toric variety corresponding to a polyhedral fan in ambient dimension 4
-```
-"""
-function Base.:*( v::AbstractNormalToricVariety, w::AbstractNormalToricVariety )
-    gap_NormalToricVariety = v.GapNTV * w.GapNTV
-    return NormalToricVariety( gap_NormalToricVariety )
-end
-export *
-
-
-struct CharacterToRationalFunction
-           gap_CharacterToRationalFunction::GapObj
-end
-export CharacterToRationalFunction
-
-
-@doc Markdown.doc"""
-    character_to_rational_function( l::Vector{Int}, v::AbstractNormalToricVariety )
-
-Turn the character `l` of the normal toric variety `v` into a rational function.
-"""
-function character_to_rational_function( l::Vector{Int}, v::AbstractNormalToricVariety )
-    gap_CharacterToRationalFunction = GAP.Globals.CharacterToRationalFunction( GapObj( l ), v.GapNTV )
-    return CharacterToRationalFunction( gap_CharacterToRationalFunction )
-end
-export character_to_rational_function
-
-
-@doc Markdown.doc"""
-    blowup_on_ith_minimal_torus_orbit( v::AbstractNormalToricVariety, i::Int )
-
-Compute the blow up of the normal toric variety `v` on the i-th minimal torus orbit.
-"""
-function blowup_on_ith_minimal_torus_orbit( v::AbstractNormalToricVariety, i::Int )
-    gap_blowup_variety = GAP.Globals.BlowUpOnIthMinimalTorusOrbit( v.GapNTV, GapObj( i ) )
-    return NormalToricVariety( gap_blowup_variety )
-end
-export blowup_on_ith_minimal_torus_orbit
-
 
 @doc Markdown.doc"""
     ith_betti_number( v::AbstractNormalToricVariety, i::Int )
 
 Compute the i-th Betti number of the normal toric variety `v`.
 """
 function ith_betti_number( v::AbstractNormalToricVariety, i::Int )
-    return GAP.Globals.ithBettiNumber( v.GapNTV, GapObj( i ) )::Int
+    if mod( i, 2 ) != 0
+        return 0
+    end
+    k = i÷2
```

Comment:
I'd recommend to avoid unicode here
```suggestion
    k = div(i, 2)
```
or
```suggestion
    k = i >> 1
```